From 4be010e18b741b4bc40f65145556657c5b59b788 Mon Sep 17 00:00:00 2001
From: Angela Cheng <angelaptcheng@gmail.com>
Date: Fri, 21 Aug 2026 06:55:39 +0000
Subject: [PATCH] PSUSensor: add MPQ82D00 PMBus device support

Add support for the MPS MPQ82D00 PMBus device.

The corresponding kernel driver is available since Linux v7.3 [1].

[1] https://github.com/torvalds/linux/commit/ff71049f36172fd3590bd170d1bc9cdd9c3e6839
---
 src/psu/PSUSensorMain.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/psu/PSUSensorMain.cpp b/src/psu/PSUSensorMain.cpp
index ff76a77..602a186 100644
--- a/src/psu/PSUSensorMain.cpp
+++ b/src/psu/PSUSensorMain.cpp
@@ -140,6 +140,7 @@ static constexpr auto sensorTypes = std::to_array<
     {"MP5998", I2CDeviceType{"mp5998", true}},
     {"MP9945", I2CDeviceType{"mp9945", true}},
     {"MP29612", I2CDeviceType{"mp29612", true}},
+    {"MPQ82D00", I2CDeviceType{"mpq82d00", true}},
     {"MPQ8785", I2CDeviceType{"mpq8785", true}},
     {"NCP4200", I2CDeviceType{"ncp4200", true}},
     {"PLI1209BC", I2CDeviceType{"pli1209bc", true}},
-- 
2.53.0-Meta

